core-processor/src/processor.rs

```diff
@@ -25,116 +25,49 @@ use crate::{
     executor,
     ext::ProcessorExt,
 };
-use alloc::{collections::BTreeMap, vec::Vec};
+use alloc::vec::Vec;
 use gear_backend_common::{Environment, IntoExtInfo};
 use gear_core::{
     env::Ext as EnvExt,
     message::{Dispatch, DispatchKind, ExitCode, Message},
     program::ProgramId,
 };
 
+enum SuccessfulDispatchResultKind {
```

Comment:
providing from/into from common dispatch result kind allows removing strange matching before `process_success` call

Reply:
> providing from/into from common dispatch result kind allows removing strange matching before `process_success` call

As I see this `From` implementation will require empty branch (`_ => ...`) processing. This yields either silencing the wrong enum option or return the `unreachable()` back.

Reply:
This is intentional since other methods require `unreachable`/`panic`/etc